src/backend/xilinx/toplevel.rs

```diff
@@ -79,6 +104,44 @@ impl Backend for XilinxInterfaceBackend {
     }
 }
 
+////I __think__ this should always be in the same order because
+//of IdList being in the same order
+//should this be a macro?
+fn external_memories_cells(
+    comp: &ir::Component,
+) -> Vec<std::rc::Rc<std::cell::RefCell<ir::Cell>>> {
+    comp.cells
+        .iter()
+        .filter(|cell_ref| {
+            matches!(cell_ref.borrow().get_attribute("external"), Some(&1))
+        })
+        .cloned()
+        .collect()
+}
+
+//returns vector of memory widths from a component
```

Comment:
You can combine these these functions into one that returns a `(WIDTH, SIZE, IDX_SIZE)` tuple for each memory instead of doing this separately in three functions